packages/std/src/os/vfs.rs

```diff
@@ -55,16 +64,109 @@ pub fn validate_username(username: String) -> Result<bool, ContractError> {
     Ok(true)
 }
 
-pub fn validate_path_name(path: String) -> Result<bool, ContractError> {
-    let re = Regex::new(PATH_REGEX).unwrap();
+pub fn validate_path_name(api: &dyn Api, path: String) -> Result<bool, ContractError> {
+    let andr_addr = AndrAddr::from_string(path.clone());
+    let is_path_reference = path.contains('/');
+    let starts_with_tilde = path.starts_with('~');
+    let includes_protocol = andr_addr.get_protocol().is_some();
+
+    // Path is of the form ~/home/... or ~username/directory
+    if is_path_reference && starts_with_tilde {
+        let re = Regex::new(PATH_REGEX).unwrap();
+        ensure!(
+            re.is_match(&path),
+            ContractError::InvalidPathname {
+                error: Some("Pathname includes an invalid character".to_string())
+            }
+        );
+
+        // let mut components = path.split('/');
```

Comment:
Is there still a need for that commented code?

Reply:
Nope, will remove